jiant/pytorch_transformers_interface/modules.py

```diff
@@ -71,46 +76,72 @@ def parameter_setup(self, args):
             # Always have one more mixing weight, for lexical layer.
             self.scalar_mix = scalar_mix.ScalarMix(self.max_layer + 1, do_layer_norm=False)
 
+    def correct_sent_indexing(self, sent):
+        """ Correct id difference between pytorch_transformers and AllenNLP.
+        The AllenNLP indexer adds'@@UNKNOWN@@' token as index 1, and '@@PADDING@@' as index 0
+        
+        args:
+            sent: batch dictionary, in which 
+                sent[self.tokenizer_required]: <long> [batch_size, var_seq_len] input token IDs
+        
+        returns:
+            ids: <long> [bath_size, var_seq_len] corrected token IDs
+            mask: <long> [bath_size, var_seq_len] sentence mask 
+        """
+        ids = sent[self.tokenizer_required]
+
+        mask = (ids != 0).long()
+        ids[ids == 0] = self._pad_id + 2
+        # map AllenNLP @@PADDING@@ to _pad_id in specific pytorch_transformer
+        if self._unk_id is not None:
+            ids[ids == 1] = self._unk_id + 2
+            # map AllenNLP @@UNKNOWN@@ to _unk_id in specific pytorch_transformer
+        ids -= 2  # shift indexes to match pretrained token embedding indexes
+
+        return ids, mask
```

Comment:
As we've already seen, this bit of code is fairly unintuitive. Do lots of print-based tests to double check this with each of the modules.